twine/commands/upload.py

```diff
@@ -257,10 +258,14 @@ def upload(dists, repository, sign, identity, username, password, comment,
         if resp.is_redirect:
             raise exc.RedirectDetected(
                 ('"{0}" attempted to redirect to "{1}" during upload.'
-                 ' Aborting...').format(config["respository"],
+                 ' Aborting...').format(config["repository"],
                                         resp.headers["location"]))
         # Otherwise, raise an HTTPError based on the status code.
-        resp.raise_for_status()
+        if resp.status_code == 400 and skip_existing:
```

Comment:
But 400 means Bad Request, which could happen for all sorts of reasons, and who knows what might come in the future that would produce 400.  I'm not saying "file exists" could produce other status codes, just that sometimes 400 might mean something other than "file exists".